tests/test_dataset.py

```diff
@@ -0,0 +1,129 @@
+# ruff: noqa: S105, S106, SLF001
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pyarrow import fs
+
+from timdex_dataset_api.dataset import DatasetNotLoadedError, TIMDEXDataset
+
+
+@pytest.mark.parametrize(
+    ("location", "expected_filesystem", "expected_source"),
+    [
+        ("/path/to/dataset", fs.LocalFileSystem, "/path/to/dataset"),
+        (
+            ["/path/to/records1.parquet", "/path/to/records2.parquet"],
+            fs.LocalFileSystem,
+            ["/path/to/records1.parquet", "/path/to/records2.parquet"],
+        ),
+        ("s3://bucket/path/to/dataset", fs.S3FileSystem, "bucket/path/to/dataset"),
+        (
+            [
+                "s3://bucket/path/to/dataset/records1.parquet",
+                "s3://bucket/path/to/dataset/records2.parquet",
+            ],
+            fs.S3FileSystem,
+            [
+                "bucket/path/to/dataset/records1.parquet",
+                "bucket/path/to/dataset/records2.parquet",
+            ],
+        ),
+    ],
+)
+@patch("timdex_dataset_api.dataset.TIMDEXDataset.get_s3_filesystem")
+def test_parse_location_single_local_directory(
+    get_s3_filesystem,
+    location,
+    expected_filesystem,
+    expected_source,
+):
+    get_s3_filesystem.return_value = fs.S3FileSystem()
+    filesystem, source = TIMDEXDataset.parse_location(location)
+    assert isinstance(filesystem, expected_filesystem)
+    assert source == expected_source
+
+
+@patch("timdex_dataset_api.dataset.fs.S3FileSystem")
+@patch("timdex_dataset_api.dataset.boto3.session.Session")
+def test_get_s3_filesystem_success(mock_session, mock_s3_filesystem):
+    mock_credentials = MagicMock()
+    mock_credentials.secret_key = "fake_secret_key"
+    mock_credentials.access_key = "fake_access_key"
+    mock_credentials.token = "fake_session_token"
+    mock_session.return_value.get_credentials.return_value = mock_credentials
+    mock_session.return_value.region_name = "us-east-1"
```

Comment:
Did you consider `moto` for this? Seems like a lot of overhead in the test for what's actually tested but I do see the symmetry with the other tests

Reply:
Great call.  This test used to do more, but once the S3 filesystem was refactored out, it's _mostly_ testing AWS connecting now.  Using moto will allow testing specifically the type of the object returned from this method.

Changing forthcoming.